### PR TITLE
JBIDE-23668: Add profiles and properties needed for running integrati…

### PR DESCRIPTION
**Pull request by rawagner**
…on tests against devstudio

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -1066,53 +1070,141 @@
...
+						<artifactId>download-maven-plugin</artifactId>
+						<executions>
+							<execution>
+								<id>download-pluginCustomization</id>
+								<phase>pre-integration-test</phase>
+								<goals>
+									<goal>wget</goal>
+								</goals>
+								<configuration>
+									<url>https://raw.githubusercontent.com/jbosstools/jbosstools-integration-tests/master/tests/pluginCustomization.ini</url>
```
Comment (rawagner) [inline parent/pom.xml]: @nickboldt This location is only temporary - used as an example. Can you suggest where to put pluginCustomization.ini file ?

Comment (nickboldt) [inline parent/pom.xml]: Since this file is only needed by integration tests for jbosstools, why would you want it anywhere but in github.com/jbosstools/jbosstools-integration-tests ?

What's wrong with https://raw.githubusercontent.com/jbosstools/jbosstools-integration-tests/master/tests/pluginCustomization.ini ?

File: parent/pom.xml [hunk 2]
```diff
@@ -153,6 +153,8 @@
 		<!-- discovery.tpc.version usually is usually set to ${tpc.version} but it
 		can happen that they need to have different values, hence 2 properties -->
 		<discovery.tpc.version>${TARGET_PLATFORM_CENTRAL_MAX}</discovery.tpc.version>
+		<testProduct>com.jboss.devstudio.core.product</testProduct>
+		<devstudio.repository>https://devstudio.redhat.com/10.0/snapshots/updates/</devstudio.repository>
```
Comment (nickboldt) [inline parent/pom.xml]: Does this URL ever change? Do you ever use the /stable/ URL, for example? If it's always /snapshots/ then I'm not sure we need it hardcoded in parent pom. Maybe set this in your jbosstools-integration-tests root pom instead?

File: parent/pom.xml [hunk 3]
```diff
@@ -153,6 +153,8 @@
 		<!-- discovery.tpc.version usually is usually set to ${tpc.version} but it
 		can happen that they need to have different values, hence 2 properties -->
 		<discovery.tpc.version>${TARGET_PLATFORM_CENTRAL_MAX}</discovery.tpc.version>
+		<testProduct>com.jboss.devstudio.core.product</testProduct>
```
Comment (nickboldt) [inline parent/pom.xml]: Please move this to line 19 along with the other global variables.

File: parent/pom.xml [hunk 4]
```diff
@@ -1066,53 +1070,136 @@
...
 		</profile>
 
+		<!-- JBIDE-23206
+			 To force a local build of a pull request to use current date/time when generating plugins and features,
+			 instead of using jgit timestamps (which could be in the past, and therefore result in stuff that Eclipse
+			 thinks is OLDER that a current N build, use this "R" profile:
+			 mvn verify -PR
+		-->
 		<profile>
-			<id>test-on-installed-app</id>
```
Comment (nickboldt) [inline parent/pom.xml]: Why did you remove the test-on-installed-app profile? And why add another R profile? There was another one on line 1102 already.

File: parent/pom.xml (hunk 2 above)
Comment (rawagner) [inline parent/pom.xml]: Usually we want to run our tests against nighly build so we are using /snapshots/ site as default. If we want to run against stable or some other specific build we can override the property. 

Plan is to move all integration-tests to component specific repository which means there will be no jbosstools-integration-tests repo and no jbosstools-integration-tests pom. That is why we need to move all this configuration to jbt parent pom.

File: parent/pom.xml (hunk 1 above)
Comment (rawagner) [inline parent/pom.xml]: same as I wrote below - Plan is to move all integration-tests to component specific repository which means there will probably be no jbosstools-integration-tests repo so in the long run we will have to move pluginCustomization.ini to some other location

File: parent/pom.xml (hunk 3 above)
Comment (rawagner) [inline parent/pom.xml]: will do

File: parent/pom.xml [hunk 5]
```diff
@@ -1066,53 +1070,136 @@
...
+						<configuration>
+							<timestampProvider>default</timestampProvider>
+						</configuration>
+					</plugin>
+				</plugins>
+			</build>
+		</profile>
+		
+		
+		<!-- IT Tests profiles -->
```
Comment (nickboldt) [inline parent/pom.xml]: IT = Integration Tests. Therefore IT Tests = Integration Tests Tests. 

Why not just call this "itests" or "integration-tests" ? Same question for the "it-tests" profile. Why not just "it" or "itests" ?

File: parent/pom.xml (hunk 4 above)
Comment (rawagner) [inline parent/pom.xml]: test-on-installed-app is basically the same profile as preinstalled-devstudio but preinstalled-devstudio has a bit updated configuration. 

Also I did not add R profile - thats just github incorrectly showing changes (take a look at file here https://github.com/rawagner/jbosstools-build/blob/a7deb18c724626bc0a0d2f5a2b181337f86ad4c4/parent/pom.xml - just one R profile )

File: parent/pom.xml (hunk 1 above)
Comment (nickboldt) [inline parent/pom.xml]: So then maybe this file should move to this repo, in github.com/jbosstools/jbosstools-build/tests/pluginCustomization.ini ?

File: parent/pom.xml (hunk 4 above)
Comment (nickboldt) [inline parent/pom.xml]: Yeah, I realized later the problem was that you reordered things and git diff was confused.

File: parent/pom.xml (hunk 5 above)
Comment (rawagner) [inline parent/pom.xml]: ok, i will change that

File: parent/pom.xml (hunk 1 above)
Comment (rawagner) [inline parent/pom.xml]: sounds good!

File: parent/pom.xml (hunk 2 above)
Comment (nickboldt) [inline parent/pom.xml]: BTW, if any of these properties need to be fed to Maven via bash commandline, I would suggest not using periods in them, as Bash doesn't understand how to deal with a ${variable.name}.

Better to use ${variableName} or ${variable_name} or even ${variablename}.

File: parent/pom.xml (hunk 1 above)
Comment (rawagner) [inline parent/pom.xml]: file added to itests folder & updated download link

File: parent/pom.xml (hunk 2 above)
Comment (rawagner) [inline parent/pom.xml]: updated all variables I've added to follow ${variableName} pattern

File: parent/pom.xml (hunk 3 above)
Comment (rawagner) [inline parent/pom.xml]: moved

File: parent/pom.xml (hunk 5 above)
Comment (rawagner) [inline parent/pom.xml]: updated to ITests

**Comment by rawagner** [comment]
PR updated

**Comment by nickboldt** [comment]
Some minor changes: https://github.com/jbosstools/jbosstools-build/pull/241/commits/f35fd9da06b0bed18fa66c58eec08cbd9ad8d371?w=1 What do you think, @rawagner  ?

**Comment by rawagner** [comment]
Looks fine. Theres just one issue - maven does not support activation of profiles based on two (or more) properties as defined here https://github.com/nickboldt/jbosstools-build/blob/f35fd9da06b0bed18fa66c58eec08cbd9ad8d371/parent/pom.xml#L1154 - pom is not parsable. We would need to remove testInstallBase property from activation block.

But then when i run "mvn help:active-profiles" all 3 ITests profiles are active - which means both preinstalled-devstudio and download-devstudio and we should really have just one of these profiles active at the same time.

**Comment by nickboldt** [comment]
What I wanted was to ensure that these profiles won't run with -DskipITests, not make them run if skipITests=false. So I guess my change won't work here. :(

As to activation based on 2 or more properties, this was fixed in Maven 3.2.2. http://stackoverflow.com/a/24176055/1661797

**Comment by rawagner** [comment]
LGTM!
